icu4c/source/test/intltest/numbertest_api.cpp

```diff
@@ -2069,6 +2102,38 @@ void NumberFormatterApiTest::sign() {
             u"-444,444.00 US dollars");
 }
 
+void NumberFormatterApiTest::signCoverage() {
+    // https://unicode-org.atlassian.net/browse/ICU-20708
+    IcuTestErrorCode status(*this, "signCoverage");
+    const struct TestCase {
+        UNumberSignDisplay sign;
+        const char16_t* expectedStrings[7];
+    } cases[] = {
+        { UNUM_SIGN_AUTO, { u"-∞", u"-1", u"-0", u"0", u"1", u"∞", u"NaN" } },
+        { UNUM_SIGN_ALWAYS, { u"-∞", u"-1", u"-0", u"+0", u"+1", u"+∞", u"+NaN" } },
+        { UNUM_SIGN_NEVER, { u"∞", u"1", u"0", u"0", u"1", u"∞", u"NaN" } },
+        { UNUM_SIGN_EXCEPT_ZERO, { u"-∞", u"-1", u"-0", u"0", u"+1", u"+∞", u"NaN" } },
+    };
+    const double inputs[] = {
+        -INFINITY, -1, -0.0, 0, 1, INFINITY, NAN
```

Comment:
NaNs have a sign bit, although it's not supposed to mean anything.
Apparently some print functions will output -NaN.
What do we do with it? Should we test it?

Reply:
Added tests and logic to handle -NaN.